framework/src/modules/token/api.ts

```diff
@@ -450,6 +458,146 @@ export class TokenAPI extends BaseAPI {
 		return chainID.equals(CHAIN_ID_ALIAS_NATIVE);
 	}
 
+	public async transferCrossChain(
+		apiContext: APIContext,
+		senderAddress: Buffer,
+		receivingChainID: Buffer,
+		recipientAddress: Buffer,
+		tokenID: Buffer,
+		amount: bigint,
+		messageFee: bigint,
+		data: string,
+	): Promise<void> {
+		if (amount < BigInt(0)) {
+			throw new Error('Amount must be greater or equal to zero.');
+		}
+		if (senderAddress.length !== ADDRESS_LENGTH) {
+			throw new Error(`Invalid sender address ${senderAddress.toString('hex')}.`);
+		}
+		if (recipientAddress.length !== ADDRESS_LENGTH) {
+			throw new Error(`Invalid recipient address ${recipientAddress.toString('hex')}.`);
+		}
+
+		const canonicalTokenID = await this._getCanonicalTokenID(apiContext, tokenID);
+		const userStore = apiContext.getStore(this.moduleID, STORE_PREFIX_USER);
+		const sender = await userStore.getWithSchema<UserStoreData>(
+			getUserStoreKey(senderAddress, canonicalTokenID),
+			userStoreSchema,
+		);
+
+		if (sender.availableBalance < amount) {
+			throw new Error(
+				`Sender ${senderAddress.toString(
+					'hex',
+				)} balance ${sender.availableBalance.toString()} is not sufficient for ${amount.toString()}`,
+			);
+		}
+		const [chainID, localID] = splitTokenID(canonicalTokenID);
+
+		const [LSK_CHAIN_ID] = splitTokenID(TOKEN_ID_LSK);
+		const possibleChainIDs = [CHAIN_ID_ALIAS_NATIVE, receivingChainID, LSK_CHAIN_ID];
+		const isAllowed = possibleChainIDs.some(id => id.equals(chainID));
+		if (!isAllowed) {
+			throw new Error(`Invalid chain id ${chainID.toString('hex')} for transfer cross chain.`);
+		}
+		let newTokenID: Buffer;
+		if (CHAIN_ID_ALIAS_NATIVE.equals(chainID)) {
+			const { id: ownChainID } = await this._interoperabilityAPI.getOwnChainAccount(apiContext);
+			newTokenID = Buffer.concat([ownChainID, localID]);
+		} else {
+			newTokenID = tokenID;
+		}
+
+		if ([CHAIN_ID_ALIAS_NATIVE, receivingChainID].some(id => id.equals(chainID))) {
+			const message = codec.encode(crossChainTransferMessageParams, {
+				tokenID: newTokenID,
+				amount,
+				senderAddress,
+				recipientAddress,
+				data,
+			});
+			const sendResult = await this._interoperabilityAPI.send(
+				apiContext,
+				senderAddress,
+				this.moduleID,
+				CROSS_CHAIN_COMMAND_ID_TRANSFER,
+				receivingChainID,
+				messageFee,
+				CCM_STATUS_OK,
+				message,
+			);
+			if (!sendResult) {
+				return;
+			}
```

Comment:
@MaximeGagnebin 
Assuming send will never throw and return the result. Handling like this, if send api fails, it doesn't subtract the amount from sender, but proceeding with cross chain transfer, is ok?